compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2577,6 +2577,8 @@ object Parsers {
       parents match {
         case parent :: Nil if !in.isNestedStart =>
           reposition(if (parent.isType) ensureApplied(wrapNew(parent)) else parent)
+        case tkn if in.token == INDENT =>
```

Comment:
With brace syntax, the last token consumed by a parser is `}`, but with indent syntax, the last token consumed by a parser is OUTDENT, which causes mismatching spans. `Template` returned from templateBodyOpt has the exact span, so we reuse it.